bio_hansel/qc/checks.py

```diff
@@ -0,0 +1,196 @@
+            else:
+                coverage_msg = 'Okay coverage depth ({:.1} >= {:.1} expected), but this may be the wrong ' \
+                               'serovar or species for scheme "{}"'.format(
+                    depth,
+                    p.low_coverage_depth_freq,
+                    st.scheme)
+            error_messages = '{status}: {p_missing:.2%} missing tiles; more than {p_missing_threshold:.2%} missing ' \
+                             'tiles threshold. {coverage_msg}'.format(
+                status=QC.MISSING_TILES_ERROR_1,
+                p_missing=p_missing,
+                p_missing_threshold=p.max_perc_missing_tiles,
+                coverage_msg=coverage_msg)
+        else:
+            error_messages = '{}: {:.2%} missing tiles; more than {:.2%} missing tile threshold'.format(
+                QC.MISSING_TILES_ERROR_1,
+                p_missing,
+                p.max_perc_missing_tiles)
+        error_status = QC.FAIL
+    return error_status, error_messages
+
+
+def is_mixed_subtype(st: Subtype, df: DataFrame, *args) -> Tuple[Optional[str], Optional[str]]:
+    """Is the subtype result mixed?
+
+    Note:
+        Check for `+` and `-` tiles that are present for the target site above minimum coverage threshold.
+        If the subtyping result came back with an inconsistent call, provide an adequate error message to the user.
+
+    Args:
+        st: Subtype results
+        df: Subtyping results dataframe
+        args: unused args
+
+    Returns:
+        None, None if not mixed subtype result; otherwise, "FAIL", error message
+    """
+    if not st.are_subtypes_consistent:
+        return QC.FAIL, \
+               '{}: Mixed subtypes found: "{}".'.format(
+                   QC.MIXED_SAMPLE_ERROR_2,
+                   '; '.join(st.inconsistent_subtypes))
+    conflicting_tiles = get_conflicting_tiles(st, df)
+    if conflicting_tiles is None or conflicting_tiles.shape[0] == 0:
+        return None, None
+
+    return QC.FAIL, '{}: Mixed subtype detected. Positive and negative tiles detected for ' \
+                    'the same target site "{}" for subtype "{}".'.format(
+        QC.MIXED_SAMPLE_ERROR_2,
+        '; '.join(conflicting_tiles['refposition'].tolist()),
+        st.subtype)
+
+
+def is_missing_too_many_target_sites(st: Subtype, df: DataFrame, p: SubtypingParams) -> Tuple[
+    Optional[str], Optional[str]]:
+    """Are there too many missing target sites for an expected subtype?
+
+    Check if there are any refpositions missing from the subtyping scheme in the result.
+    If there are missing refpositions, this is a missing target.
+    If there are too many missing targets, this result is an Ambiguous Result.
+
+    Args:
+        st: Subtype results
+        df: Subtyping results dataframe
+        p: Subtyping/QC parameters
+
+    Returns:
+        None, None if less missing targets than tolerated; otherwise, "FAIL", error message
+    """
+    if not st.are_subtypes_consistent:
```

Comment:
```python
if not st.are_subtypes_consistent or st.subtype is None:
    # Early exit
    return None, None
```

Reply:
The `st.subtype is None` check should be done before any of the QC check functions. But it's probably a good idea to recheck just in case.